doc/userguide/devguide/codebase/installation-from-git.rst

```diff
@@ -0,0 +1,168 @@
+
+Add ``${HOME}/.cargo/bin`` to your path.
+
+.. code-block:: bash
+
+  export PATH=$PATH:${HOME}/.cargo/bin
+  cargo install --force cbindgen
+
+Depending on the current status of your system, it may take a while to
+complete this process.
+
+**IPS**
+
+By default, Suricata works as an IDS. If you want to use it as a IDS and IPS
+program, enter:
+
+.. code-block:: bash
+
+  sudo apt-get -y install libnetfilter-queue-dev libnetfilter-queue1 \
+  libnfnetlink-dev libnfnetlink0
+
+Suricata
+~~~~~~~~
+
+First, it is convenient to create a directory for Suricata.
+Name it 'suricata' for example. Open the terminal and enter:
+
+.. code-block:: bash
+
+  mkdir suricata
+
+
+Followed by:
+
+.. code-block:: bash
+
+  cd suricata
+
+Next, enter the following line in the terminal:
+
+.. code-block:: bash
+
+  git clone git@github.com:OISF/suricata.git
+  cd suricata
+
+Libhtp is not bundled. Get libhtp by doing:
+
+.. code-block:: bash
+
+  ./scripts/bundle.sh libhtp
+
+Followed by:
+
+.. code-block:: bash
+
+  ./autogen.sh
+
+
+To configure, please enter:
+
+.. code-block:: bash
+
+  ./configure
+
+
+To compile, please enter:
+
+.. code-block:: bash
+
+  make
+
+To install Suricata, enter:
+
+.. code-block:: bash
+
+  sudo make install
+  sudo ldconfig
+
+To install suricata-update
+
+Follow the instructions found
+https://suricata-update.readthedocs.io/en/latest/quickstart.html
+
+.. note:: If you would like to build ``suricata-update`` from source, enter:
+
+  .. code-block:: bash
+
+    ./scripts/bundle.sh suricata-update
+    cd suricata-update
+    python setup.py build
+    sudo python setup.py install
+    sudo suricata-update
+
+Auto setup
+~~~~~~~~~~
+
+You can also use the available auto setup features of Suricata:
+
+
+ex:
+
+.. code-block:: bash
+
+  ./configure && make && sudo make install-conf
+
+*make install-conf*
+would do the regular "make install" and then it would automatically
+create/setup all the necessary directories and ``suricata.yaml`` for you.
+
+.. code-block:: bash
+
+  ./configure && make && make install-rules
+
+*make install-rules*
+would do the regular "make install" and then it would automatically download
+and set up the latest ruleset from Emerging Threats available for Suricata
+
+.. code-block:: bash
+
+  ./configure && make && make install-full
+
+*make install-full*
+would combine everything mentioned above (install-conf and install-rules) -
+and will present you with a ready to run (configured and set up) Suricata
+
+
+Please continue with
+`Basic Setup <https://redmine.openinfosecfoundation.org/projects/suricata/
+wiki/Basic_Setup>`_.
+
+In case you have already made a map for the most recent code, download the
+code into that map, and want to download recent code again, please enter:
+
+.. code-block:: bash
+
+  cd suricata/suricata
+
+next, enter:
+
+.. code-block:: bash
+
+  git pull
+
+After that, you start again at running *./autogen.sh*.
```

Comment:
Let's also reword this:

After that, you should run ``./autogen.sh`` again.

Reply:
Just so I'm sure, is running `./autogen.sh` the only required step after pulling new changes? I thought there is a need to run `./configure`, `make` and other required command to re-install Suri. I'm not sure the reworded sentence shows this (unless I'm mistaken — English :( )

Reply:
Oh...disregard my comment. There is an echo statement detailing the next step in `./autogen.sh`